gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/io/binary/types/VertexPropertySerializer.java

```diff
@@ -66,7 +66,10 @@ protected void writeValue(final VertexProperty value, final Buffer buffer, final
         // we don't serialize the parent vertex, let's hold a place for it
         context.write(null, buffer);
 
-        final List<?> asList = IteratorUtils.toList(value.properties());
-        context.write(asList, buffer);
+        // Neo4jVertexProperty throw UnsupportedOperationException for properties,
+        // so for now VertexProperty will be serialized without inner properties
+        // final List<?> asList = IteratorUtils.toList(value.properties());
+        // context.write(asList, buffer);
+        context.write(null, buffer);
```

Comment:
I assume that this is temporary? not all graphs support meta/multiproperties so this is a likely to be a general problem. if you had the `Graph` or `g` you could test `features()` to see what the graph supported and then either write the metaproperties or not, but i don't think you have access to that here. 

Any idea why this works for GraphSON without omitting the metaproperties?

Reply:
GraphSON uses workaround by getting properties starting from `keys`. It only works when `Neo4jVertexProperty.vertexPropertyNode` is null or empty.

Reply:
what is the plan to re-enable this?

Reply:
I can do it like in GraphSON, but then there will be a risk of getting a problem if some provider wants to use `vertexPropertyNode`.
Do you have other suggestions?

Reply:
I'm not sure i understand the nature of the risk you're describing, but the `keys()` implementation for `Neo4jGraph` is strange anyway. It returns an empty `Set`  while `properties()` throws an exception...that's pretty inconsistent.  I"m not even sure `vertexPropertyNode` is relevant - i have a feeling it is dead code and `keys()` was missed in some refactoring at some point.

Neo4j really isn't the issue though. The real issue is how it works for any `Graph` implementation that doesn't support metaproperties. As the current approach has worked for years in GraphSON/Gryo I feel like we can fairly assume it should be safe for GraphBinary. 

That said, perhaps we should do better than assume. You could encode the idea that `properties()` / `keys()` should not throw exceptions but instead return empty iterables.  You'd do that by modifying test semantics for `FeatureSupportTest` and probably `VertexPropertyTest` (there might be others that fail as a result of this change). This change would require Upgrade Docs for providers to take note of. I think you'd want to wholly remove `VertexProperty.Exception.metaPropertiesNotSupported` basically and see what breaks.

Reply:
`VertexProperty.Exception.metaPropertiesNotSupported` used when `VertexProperty` doesn't support getting and setting meta-properties. I changed getter behavior to return empty Iterable, but kept Exception for setter.